### ISSUE TITLE
Promtail does not use the http_path_prefix, metrics works but the promtail UI does not.

### ISSUE THREAD

**Issue by iangregsondev** [grafana/loki#4765]
**Describe the bug**
The bug was described here https://github.com/grafana/loki/issues/1709 but it was closed by the stale bot, but it should not have as it still is a problem. Others are chatting there and it was requested to open up a new ticket.

This is not stale.  To summarise.

I am using    ```http_path_prefix /promtail```

If you go to http://some-domain.com/promtail/metrics - all is well, seems to return the metrics as it should and you can see the prefix.

but do - http://some-domain.com/promtail - which should bring up the promtail UI - it doesn't because of a redirect to "/targets", this should respect the http_path_prefix.

Even if you ignore that and go to http://some-domain.com/promtail/targets then it displays a broken UI, missing CSS I assume because it couldn't load because of the path_prefix not being followed.

![image](https://user-images.githubusercontent.com/2370364/141995623-f46528ce-331b-4e68-bdf3-4a25601f4a33.png)

I use the server for many things on http://some-domain.com - and I don't want to create a subdomain, it would get out of control quickly :-)

**To Reproduce**
See above

I am using caddy in front of promtail, here is the config in the caddfile, for transparency

```
some-domain.com {
        route /promtail/* {
           reverse_proxy localhost:9080
        }
}
```

**Expected behavior**
http_path_prefix should be used on promtail - all routes, redirects etc. Maybe this is easy to solve with relative HTML routes.

**Environment:**
promtail 2.4.1

**Screenshots, Promtail config, or terminal output**
See above.

**Comment by iangregsondev** [comment]
@taisho6339 @kavirajk @cyriltovena Hi, I have just installed v 2.4.2 - i presume the fix is in there ?

But its still the same..

Can we reopen this, instead of creating a new issue ?

```
server:
  grpc_listen_port: 9081
  http_listen_port: 9080
  http_path_prefix: /promtail/
```

and if I go to my domain...

Going to promtail seems to redirect me to "/" but it should be /promtail

and if I force it to to open /promtail/targets

then the CSS etc is still messed up, if you check the chrome dev tools, you can see its reading it from the wrong place

**Comment by taisho6339** [comment]
In my understanding, v2.4.2 doesn't contain this fix.

**Comment by iangregsondev** [comment]
Ahh - that makes sense then :-) lol...

The thing is that it was merged a while back and then I noticed you just released a new version - so my assumption was that it made the release..

No worries..

Do you know when it is going to release ?

Thanks in advance.

**Comment by iangregsondev** [comment]
I can confirm this appears to be released and working on promtail 2.5.0

Thanks
